src/parser_json.c

```diff
@@ -398,10 +398,22 @@ json_get_anydata(struct lyd_node_anydata *any, const char *data)
     do {
         switch (data[len]) {
         case '{':
-            c++;
+            if (skip == 0) {
+                c++;
+            }
             break;
         case '}':
-            c--;
+            if (skip == 0) {
+                c--;
+            }
+            break;
+        case '\\':
+            if (skip == 1 && data[len + 1]) {
```

Comment:
Replace `skip == 1` with just `skip` and please add a comment with an explanation. Something like `when parsing a string, ignore escaped quotes to not end it prematurely`.